graph-builder/src/graph.rs

```diff
@@ -208,61 +190,23 @@ pub async fn run(settings: &config::AppSettings, state: &State) -> ! {
 
         debug!("graph update triggered");
         let scrape_timer = UPSTREAM_SCRAPES_DURATION.start_timer();
-        let scrape = registry::fetch_releases(
-            &registry,
-            &settings.repository,
-            username.as_ref().map(String::as_ref),
-            password.as_ref().map(String::as_ref),
-            &mut cache,
-            &settings.manifestref_key,
-            settings.fetch_concurrency,
-        )
-        .await;
-        UPSTREAM_SCRAPES.inc();
 
-        let releases = match scrape {
-            Ok(releases) => {
-                if releases.is_empty() {
-                    warn!(
-                        "could not find any releases in {}/{}",
-                        &registry.host_port_string(),
-                        &settings.repository
-                    );
-                };
-
-                // Record scrape duration
-                scrape_value = scrape_timer.stop_and_discard();
-                releases
-            }
-            Err(err) => {
-                UPSTREAM_ERRORS.inc();
-                err.iter_chain()
-                    .for_each(|cause| error!("failed to fetch all release metadata: {}", cause));
-                continue;
-            }
-        };
-        GRAPH_UPSTREAM_RAW_RELEASES.set(releases.len() as i64);
-
-        let graph = match create_graph(releases) {
-            Ok(graph) => graph,
-            Err(err) => {
-                err.iter_chain().for_each(|cause| error!("{}", cause));
-                continue;
-            }
-        };
-
-        let graph = match cincinnati::plugins::process(
+        let scrape = cincinnati::plugins::process(
             state.plugins.iter(),
             cincinnati::plugins::PluginIO::InternalIO(cincinnati::plugins::InternalIO {
-                graph,
+                // the first plugin will produce the initial graph
+                graph: Default::default(),
                 // the plugins used in the graph-builder don't expect any parameters yet
                 parameters: Default::default(),
             }),
         )
-        .await
-        {
-            Ok(internal_io) => internal_io.graph,
+        .await;
+        UPSTREAM_SCRAPES.inc();
+
+        let (graph, _) = match scrape {
```

Comment:
`internal_io.parameters` is being returned here and then thrown away?

Reply:
Yes, awesome right? :-)

Reply:
I have changed this to pass on the internal_io instance, so the outer code can finally decide which fields it needs from it.